cmd/oras/internal/display/metadata/model/discover.go

```diff
@@ -15,19 +15,54 @@ limitations under the License.
 
 package model
 
-import ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+import (
+	"fmt"
 
-type discover struct {
-	Manifests []Descriptor `json:"manifests"`
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// Discover is a model for discovered referrers.
+type Discover struct {
+	name  string
+	nodes map[digest.Digest]*Node
+	Root  *Node
+}
+
+// Node represents a node in the discovered reference tree.
+type Node struct {
+	Descriptor
+	Referrers []*Node `json:"manifests,omitempty"`
```

Comment:
This PR conflicts with #1649. Let's wait for #1649 first.

Reply:
#1649 is merged already.